### DISCUSSION TITLE
Welcome to DCS-World-Dedicated-Server-Docker Discussions!

### DISCUSSION THREAD

**Discussion by Aterfax** [Aterfax/DCS-World-Dedicated-Server-Docker#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tspindler-cms** [comment]
Hi,

I stumbled over your project a couple of months ago when I wondered how I can utilize my Linux server for DCS. It had docker running already, so setup and everything was a matter of a few minutes, maybe an hour at most.
I would be interested to learn how to automate some DCS things, like mission restart or server restart, from the command line or ultimately a discord based bot. As with many other servers, the TeTeT Dedi Test Server is not very often frequented and the mission on it becomes stale. Offering folks a possibility to restart the mission w/o giving them full VNC access would be a great feature!

Thanks for putting the effort together to push this nice project ahead!

Cheers,
TeTeT

**Reply by Aterfax** [reply]
Hi TeTeT,

Glad to hear people have been getting some use from the docker image!

In terms of automation of some server things, there is as they say, many ways to skin a cat so take my suggestions with a pinch of salt as there may be better ways of going about things.

### Mission / server restart

On mission restart or server restart, depending on what / why you need to do the context will probably denote a "best method":

- You can create or edit a mission to add a trigger to restart after a certain amount of time - https://wiki.hoggitworld.com/view/Timed_Restart
- You could make a CRON job on the host machine to restart the docker container e.g. https://www.warp.dev/terminus/docker-restart-container#restart-containers-every-hour
- You could apply the [Linux Server CRON docker mod ](https://github.com/linuxserver/docker-mods/tree/universal-cron) to enable CRON inside the docker container and set a CRON job to periodically kill the server and let the daemon restart it for you.
  - e.g. run the command ``kill $(pgrep -u abc -f "/config/.wine/drive_c/Program Files/Eagle Dynamics/DCS World Server/bin/DCS_server.exe")``
- When integration of the DCSServerBot is complete you can look at using it with the plugin scheduler: https://github.com/Special-K-s-Flightsim-Bots/DCSServerBot/blob/master/plugins/scheduler/README.md
   -  This isn't ready to go as I need to finish / convert this PR to Docker Mod format - https://github.com/Aterfax/DCS-World-Dedicated-Server-Docker/pull/35)
   
Eventually I will likely add some native functionality within the Docker image which allows for more straightforward restarts or reloads of missions via the use of the [Linux Server CRON docker mod ](https://github.com/linuxserver/docker-mods/tree/universal-cron).

n.b. if your server is auto starting but not loading and starting the mission, you will need to adjust the server settings in the webGUI:

1. Settings cog.
2. Advanced.
3. Resume on load.

![image](https://github.com/Aterfax/DCS-World-Dedicated-Server-Docker/assets/8648058/b8cda0e1-1255-4c2d-97d1-3409c05183a8)


### Restart the mission / server w/o giving people full VNC access

Originally this was the motivation behind me [filing the issue](https://github.com/Aterfax/DCS-World-Dedicated-Server-Docker/issues/31) to get compatibility with [Pterodactyl](https://pterodactyl.io/) but this is also the reason why I was planning on implementing DCS Serverbot (fantastic work by SpecialK) because it supports a huge variety of plugins and allows administration from Discord servers.

I am planning on eventually getting the PR for DCS Serverbot finished at which point hopefully the vast majority of its supported plugins will also work.

I highly recommend having a look at the functionality of DCS Serverbot as I bet it will meet the needs of most folk - https://github.com/Special-K-s-Flightsim-Bots/DCSServerBot

I'm hoping I can put in a solid chunk of work getting this integrated into the image this weekend but I cannot promise any firm timelines.

Ty!

**Comment by tspindler-cms** [comment]
Hi Aterfax,

Thanks for your thorough reply! As there's no rush with this topic for me, I'll happily wait for the DCS Serverbot integration. It sounds like a great solution, as we use Discord for chat comms anyway.

Cheers,
TeTeT

**Comment by MartinFlower21** [comment]
Hi. This is a great idea.

Im getting the error:
 Error: File '/config/.wine/drive_c/users/abc/Saved Games/DCS.server/Config/network.vault' does not exist. Cannot autostart without user logging in and saving credentials for autologin. Exiting.

I dont get the login box as per your video. I do get pop up boxes saying dcs application was modified. 

any thoughts?

**Reply by MartinFlower21** [reply]
Update: I think I have sorted it. In the env variables in Kubernetes I had not set WINEARCH=64 and WINPREFIX to my .wine path

**Reply by MartinFlower21** [reply]
I was wrong. It gets further - it now runs updater but I still get the network.vault' does not exist error. no login window opens

**Comment by Aterfax** [comment]
Disable automatic start up of the game server in any environment variables.Start the docker manually then go to the web UI,  start the game server manually from the shortcut, then login window should pop up eventually.It may pop up underneath another window, just right click on the login entry in the taskbar and either move or maximize then log in.If you tell the server to do an auto start and you haven't logged in at least once it will automatically kill itself and probably terminate the docker container.On 29 Jul 2024 10:21, MartinFlower21 ***@***.***> wrote:
Hi. This is a great idea.
Im getting the error:
Error: File '/config/.wine/drive_c/users/abc/Saved Games/DCS.server/Config/network.vault' does not exist. Cannot autostart without user logging in and saving credentials for autologin. Exiting.
I dont get the login box as per your video. any thoughts?

**Reply by MartinFlower21** [reply]
will do thanks

**Comment by MartinFlower21** [comment]
This is such a great project.

Is there any chance of a more detailed tutorial on getting retribution to work?

**Reply by Aterfax** [reply]
If you're after the instructions on using the docker mod provided with this project then some additional documentation could potentially be generated though I am unsure what is currently required.


If the need is largely around making general usage of the DCS Retribution mod, then the project's documentation and tutorials should be used: https://github.com/dcs-retribution/dcs-retribution/wiki/

**Comment by dmeador** [comment]
Hi Aterfax,  
I got the Docker DCS server working.  Thanks for sharing this!  I wanted to run SRS Server in docker as well.  
Update: I was able to get SR-Server.exe from SRS working in webtop!  It requires dotnet and d3d to vulkan support under wine, which actually works in 64-bit mode.  So, then the remainder of the work would be to add this as a docker-mod in the main project.

I would be interested in collaborating with you to get SRS supported as a docker mod... I am unfamiliar with the docker modding and need some help to spin up on that infrastructure.

**Reply by Aterfax** [reply]
Short answer is this is probably possible but only via rebasing the image onto another version that does support 32 bit.

I could target Ubuntu for this but I originally avoided the fuller fat images to try to stop this container from ballooning in size.

**Reply by dmeador** [reply]
> Short answer is this is probably possible but only via rebasing the image onto another version that does support 32 bit.
> 
> I could target Ubuntu for this but I originally avoided the fuller fat images to try to stop this container from ballooning in size.

Update, I was mistaken about needing wine-32 for SRS, I got SRS working and using winetricks to install dotnet dependencies.

**Reply by Aterfax** [reply]
Can you write down the required dotnet packages and the process you used to get it working? We can then look at getting this into a DockerMod.

Inspiration can be taken from:

https://github.com/Aterfax/DCS-World-Dedicated-Server-Docker/blob/main/docker/Dockerfile.DockerMod.dcs-retribution

This should give an idea of what is needed in terms of creating a new S6 service and ensuring the dependencies are in place to get the installation vs start up order correct.

**Reply by dmeador** [reply]
> Can you write down the required dotnet packages and the process you used to get it working? We can then look at getting this into a DockerMod.
```

winetricks dotnet48
winetricks dxvk
# that's all that's needed for SRS 
# then run server (I expect that server is the typical case)
wine SR-Server.exe
#or
# wine SR-Client.exe
```
There are a few dialogs that popup which might throw a wrench into automated install, but I think I saw some fancy wine scripting in the source that can look for windows and send clicks and key-input on window fields.

**Comment by wendelcrow** [comment]
I havent really started looking at this yet, but has anyone tried deploying this in a "proper" kubernetes cluster?
